nixos/modules/config/zram.nix

```diff
@@ -92,6 +92,46 @@ in
           as there's no gain from keeping them in RAM.
         '';
       };
+
+      use-optimal-sysctls = lib.mkOption {
+        default = true;
+        type = lib.types.bool;
+        description = ''
+          Applies some sysctls to optimize the virtual memory subsystem
+          for zramSwap.
+
+          - `vm.swappiness = 150` increase swapping aka compression to be
+            able to cache more file page data
+          - `vm.watermark_boost_factor = 0` watermark boosting can cause unpredictable stalls as seen here:
+            https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1861359
+          - `vm.watermark_scale_factor = 125` initiate kswapd much earlier
+            as zram will also apply pressure when requesting memory for
+            compressed swap
+          - `vm.page-cluster = 0` disables page prefetching
+
+          This option when enabled, sets these sysctls when you also
+          enable the zramSwap module.
+
+          Defaults to `true`.
+        '';
+      };
```

Comment:
Not sure if it's wise to pack several "optimizations" (they are not by default for a reason) together, and setting such initial values.

Reply:
Hmm yeah. The reason why they are not the defaults is because the default is assuming that someone uses a swap partition or swapfile on a disk. That was the common setup almost all distros defaulted to one and two decades ago.

Do you think these should be split up? Or should we just flip `default = false` ?